persistence/src/main/scala/akka/persistence/kafka/journal/BaseKafkaJournal.scala

```diff
@@ -0,0 +1,243 @@
+package akka.persistence.kafka.journal
+
+import akka.actor.ActorSystem
+import akka.persistence.journal.AsyncWriteJournal
+import akka.persistence.{AtomicWrite, PersistentRepr}
+import cats.Parallel
+import cats.effect._
+import cats.implicits._
+import com.evolutiongaming.catshelper.{FromFuture, Log, LogOf, ToFuture}
+import com.evolutiongaming.kafka.journal._
+import com.evolutiongaming.kafka.journal.util.CatsHelper._
+import com.evolutiongaming.kafka.journal.util.PureConfigHelper._
+import com.evolutiongaming.scassandra.CassandraClusterOf
+import com.evolutiongaming.smetrics.MeasureDuration
+import com.typesafe.config.Config
+import pureconfig.ConfigSource
+
+import scala.collection.immutable.Seq
+import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContextExecutor, Future}
+import scala.util.Try
+
+abstract class BaseKafkaJournal[A](config: Config) extends AsyncWriteJournal { actor =>
```

Comment:
why abstract ?

Reply:
Cause it is parametrized in payload type and I don't have any way to provide it with
 EventSerializer and JournalRead/Write. am I missing some other way to do it?

Reply:
what do you think about `KafkaJournal` as base with two children `KafkaJournalCirce` & `KafkaJournalPlayJson` ? and the `KafkaJournal` is not abstract but have some empty/default maybe binary only implementation?

Reply:
Changed it to KafkaJournal, removed abstract and implemented it in terms of existing Payload implementation (rebased initial commit).
Did not introduce `KafkaJournalPlayJson`, since currently we do not have a play-json-only implementation of journal, I think it's better to do it as part of a separate refactoring (extract play-json from existing Payload hierarchy and introduce separate play-json kafka journal implementation, and this implementation becomes binary only)